libdnf/repo/solv_repo.cpp

```diff
@@ -84,7 +84,8 @@ bool SolvRepo::can_use_solvfile_cache(fs::File & solvfile_cache) {
     int dnf_solv_userdata_len_read;
 
     int ret_code = solv_read_userdata(solvfile_cache.get(), &dnf_solv_userdata_read, &dnf_solv_userdata_len_read);
-    std::unique_ptr<SolvUserdata> solv_userdata(reinterpret_cast<SolvUserdata *>(dnf_solv_userdata_read));
+    std::unique_ptr<SolvUserdata, decltype(&free)> solv_userdata(
+        reinterpret_cast<SolvUserdata *>(dnf_solv_userdata_read), &free);
```

Comment:
Libsolv gets memory by `solv_malloc`! Yes, `solv_malloc` uses `malloc` internally. However, this is an implementation detail of libsolv.
Memory allocated with `solv_malloc` should be freed with `solv_free`.

Reply:
My mistake. I have updated it.
Thanks!